src/providers/bedrock/chatComplete.ts

```diff
@@ -9,613 +9,218 @@ import {
-                      const mediaType = mediaTypeParts[1];
-                      transformedMessage.content.push({
-                        type: 'image',
-                        source: {
-                          type: 'base64',
-                          media_type: mediaType,
-                          data: base64Image,
-                        },
-                      });
-                    }
-                  }
-                }
-              });
-              messages.push(transformedMessage as Message);
-            } else if (msg.role === 'tool') {
-              // even though anthropic supports images in tool results, openai doesn't support it yet
-              messages.push(transformToolMessageForAnthropic(msg));
-            } else {
-              messages.push({
-                role: msg.role,
-                content: msg.content,
-              });
-            }
+      transform: (params: BedrockChatCompletionsParams) => {
+        if (!params.messages) return [];
+        return params.messages
+          .filter((msg) => msg.role !== 'system')
+          .map((msg) => {
+            return {
+              role: msg.role === 'assistant' ? 'assistant' : 'user',
+              content: getMessageContent(msg),
+            };
           });
-        }
-
-        return messages;
       },
     },
     {
       param: 'system',
       required: false,
-      transform: (params: Params) => {
-        let systemMessage: string = '';
-        // Transform the chat messages into a simple prompt
-        if (!!params.messages) {
-          params.messages.forEach((msg) => {
-            if (
-              msg.role === 'system' &&
-              msg.content &&
-              typeof msg.content === 'object' &&
-              msg.content[0].text
-            ) {
-              systemMessage = msg.content[0].text;
-            } else if (
-              msg.role === 'system' &&
-              typeof msg.content === 'string'
-            ) {
-              systemMessage = msg.content;
-            }
-          });
-        }
-        return systemMessage;
+      transform: (params: BedrockChatCompletionsParams) => {
+        if (!params.messages) return;
+        const systemMessages = params.messages.reduce(
+          (acc: { text: string }[], msg) => {
+            if (msg.role === 'system')
+              return acc.concat(...getMessageTextContentArray(msg));
+            return acc;
+          },
+          []
+        );
+        if (!systemMessages.length) return;
+        return systemMessages;
+      },
+    },
+    {
+      param: 'inferenceConfig',
+      transform: (params: BedrockChatCompletionsParams) => {
+        return {
+          maxTokens: params.max_tokens || params.max_completion_tokens,
+          stopSequences:
+            typeof params.stop === 'string' ? [params.stop] : params.stop,
+          temperature: params.temperature,
+          topP: params.top_p,
+        };
+      },
+    },
+    {
```

Comment:
Previously, we had individual mapping for some provider specific params like anthropic_version. With converse API, these needs to go into  `additionalModelRequestFields`. But from user perspective, we should keep it backward compatible. So they can still send anthropic_version and gateway should map it to additional fields